src/Accelerators/NNPA/Transform/ZHigh/FusionOpStickUnstick.cpp

```diff
@@ -0,0 +1,393 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===-------- FusionOpStickUnstick.cpp - Fuse compute op  -----------------===//
+//
+// Copyright 2023 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This pass detects patterns of unstick -> op -> stick (or subset) and when
```

Comment:
It looks like this can be done by the current code in onnx-mlir, which is in the layout propagation pass in zHigh. It is disabled now, but you can enable it and see check if does the same thing or not: https://github.com/onnx/onnx-mlir/blob/main/src/Accelerators/NNPA/Transform/ZHigh/ZHighLayoutPropagation.cpp#L345.

In ZHighLayoutPropagation, `unstick -> op -> stick` is rewritten by:
- propagating `unstick` down, so we have `op -> unstick -> stick`, then
- `unstick -> stick` becomes identity and is removed.

Reply:
So you are suggesting that I move the functionality of FusionOpStickUnstick into the current ZHighLayoutPropagation optimization? Just moving the code there, correct?
I support a few more cases, for example arbitrary mixture of Ztensor and normal tensors for inputs/ouput.
Also because I generate simd code only, I need to know at compile time if there is a broadcast in the innermost condition or not. It cannot be a runtime decision.

Reply:
Tried in a new PR but lead to problems, explained both in that other PR and below when re-opening this PR. I now removed the "dead code" as it would result in similar issues if re-activated.